client/state/comments/reducer.js

```diff
@@ -50,15 +50,35 @@ export const deconstructStateKey = key => {
 	return { siteId: +siteId, postId: +postId };
 };
 
+const isCommentManagementEdit = newProperties =>
+	has( newProperties, 'commentContent' ) &&
+	has( newProperties, 'authorDisplayName' ) &&
+	has( newProperties, 'authorUrl' );
+
 const updateComment = ( commentId, newProperties ) => comment => {
 	if ( comment.ID !== commentId ) {
 		return comment;
 	}
 	const updateLikeCount = has( newProperties, 'i_like' ) && isUndefined( newProperties.like_count );
 
+	// Comment Management allows for modifying nested fields, such as `author.name` and `author.url`.
+	// Though, there is no direct match between the GET response (which feeds the state) and the POST request.
+	// This ternary matches and formats the updated fields sent by Comment Management's Edit form,
+	// in order to optimistically update the state without temporary loss of information.
+	const newComment = isCommentManagementEdit( newProperties )
+		? {
+				...comment,
+				author: {
+					...comment.author,
+					name: newProperties.authorDisplayName,
+					url: newProperties.authorUrl,
+				},
+				content: newProperties.commentContent,
+			}
```

Comment:
This weird indentation was caused by the automatic Prettier on commit.
It throws an eslint warning, but I'm not sure how I'm supposed to fix it.
I don't feel like removing the `/** @format */` marker, so should I change this into an if/else?